station-dispenser-api-v0/src/main/java/net/modificationstation/stationapi/api/item/DispenseUtil.java

```diff
@@ -0,0 +1,132 @@
+package net.modificationstation.stationapi.api.item;
+
+import net.minecraft.block.entity.DispenserBlockEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+
+import java.util.Random;
+
+/**
+ * Utility class for changing dispenser behavior, providing tools to customize behavior
+ * @author matthewperiut
+ * @see DispenseEvent
+ * @see CustomDispenseBehavior
+ */
+public class DispenseUtil {
+    public World world;
+    public DispenserBlockEntity dispenserBlockEntity;
+    public ItemStack itemStack;
+    private ItemStack shotItemStack;
+    public final ItemStack[] inventory;
+    public final int slot;
+    public Random random = new Random();
+    public int x;
+    public int y;
+    public int z;
+    public byte xDir = 0;
+    public byte zDir = 0;
+    public double xVel;
+    public double yVel;
+    public double zVel;
```

Comment:
What's the reason for some of these not being final? Such as `world`, for example.

Reply:
head empty lol